resources/design-document.md

```diff
@@ -1,75 +1,197 @@
-
-## _Project Title_ Design
+## PopStock Design
 
 ## 1. Problem Statement
 
-_Explain clearly what problem you are trying to solve._
+A new warehouse can be difficult to spin up, requiring you to make difficult decisions about what items you are going to stock. PopStock will solve this problem by using data to choose the most profitable items to stock, based on metrics such as margin, time-to-replenish, sales volume, item synergy, and region adjusted shipping costs.  Once a new warehouse has been spun up, users will be able to manage the inventory and generate reports.
+
 
 ## 2. Top Questions to Resolve in Review
 
-_List the most important questions you have about your design, or things that you are still debating internally that you might like help working through._
+1. Where would caching be most useful
 
-1.
-2.
-3.
 
 ## 3. Use Cases
 
 _This is where we work backwards from the customer and define what our customers would like to do (and why). You may also include use cases for yourselves (as developers), or for the organization providing the product to customers._
 
-U1. _As a [product] customer, I want to `<result>` when I `<action>`_
+U1. As a PopStock customer, I want to create new inventory items with attributes and metrics that can be used to stock warehouses
 
-U2. _As a [product] customer, I want to view my grocery list when I log into the grocery list page_
+U2. As a PopStock customer, I want to create a new warehouse with a name and a regional location
 
-U3. ...
+U3. As a PopStock customer, I want to populate a new warehouse with items selected automatically based on profitablility
 
-## 4. Project Scope
+U4. As a PopStock customer, I want to generate a report showing inventory statistics and suggesting more profitable items available in the inventory items list
 
-_Clarify which parts of the problem you intend to solve. It helps reviewers know what questions to ask to make sure you are solving for what you say and stops discussions from getting sidetracked by aspects you do not intend to handle in your design._
+U5. As a PopStock customer, I want to update the amount of available stock for each item.
 
-### 4.1. In Scope
+U6. As a PopStock customer, I want to remove items from my warehouse if no stock is currently available
 
-_Which parts of the problem defined in Sections 1 and 2 will you solve with this design? This should include the base functionality of your product. What pieces are required for your product to work?_
+U7. As a PopStock customer, I want to add items to my warehouse when space is available
 
-_The functionality described above should be what your design is focused on. You do not need to include the design for any out of scope features or expansions._
+U8. As a PopStock customer, I want to see reports on item shipments over a given time-period
```

Comment:
The idea here is that as the inventory of a warehouse is updated, it will generate shipping entries with the item, timestamp, stock change amount, etc.  There will be an ID associated with it for the customer, but that will be manually input for now. (A stretch goal would be to have a separate table and endpoints to be able to input customers/suppliers).

After looking at your feedback and giving further consideration to my table/endpoint design, I'm going to update the "shipped" table to "transactions"  and track suppliers as well, with a generated "shipment_id" and a map of items and modification count